exasol_data_science_utils_python/schema/exasol_identifier.py

```diff
@@ -0,0 +1,28 @@
+import unicodedata
+from abc import ABC, abstractmethod
+
+from typeguard import typechecked
+
+
+class ExasolIdentifier(ABC):
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def quoted_name(self) -> str:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def fully_qualified(self) -> str:
+        raise NotImplementedError()
```

Comment:
could be property?

Reply:
thought about it as well, but does it really fit for property, because it generates a value

Reply:
done